contracts/EpochTokenLocker.sol

```diff
@@ -35,6 +35,9 @@ contract EpochTokenLocker {
     // User => Token => BalanceState
     mapping(address => mapping(address => BalanceState)) private balanceStates;
 
+    // user => token => batchId => wasCredited
+    mapping(address => mapping (address => mapping(uint => bool))) public hasCreditedBalance;
```

Comment:
Would it be slightly more efficient to just store `user => token => largest_credited_batchId` ?

Reply:
I think in the current solution, we effectively only store a boolean, and the user, token, batchID are just getting hashed.

This should be more efficient than storing a uint for the batchID